processing/src/main/java/io/druid/query/extraction/LookupReferencesManager.java

```diff
@@ -128,6 +134,40 @@ public void put(Map<String, LookupExtractorFactory> lookups)
   }
 
   /**
+   * Add or update a lookup factory
+   *
+   * @param lookupName             The name of the lookup
+   * @param lookupExtractorFactory The factory of the lookup
+   *
+   * @return True if the lookup was updated, false otherwise
+   *
+   * @throws IllegalStateException if start of the factory fails
+   */
+  public boolean updateIfNew(String lookupName, final LookupExtractorFactory lookupExtractorFactory)
+  {
+    boolean updated = false;
+    synchronized (lock) {
+      assertStarted();
+      final LookupExtractorFactory prior = lookupMap.get(lookupName);
+
+      if (!lookupExtractorFactory.equals(prior)) {
+        if (!lookupExtractorFactory.start()) {
+          throw new ISE("Could not start [%s]", lookupName);
+        }
+        lookupMap.put(lookupName, lookupExtractorFactory);
+        updated = true;
+      }
+
+      if (prior != null) {
```

Comment:
Not true, but it will close when it shouldn't. Realized I missed this method in adding unit tests. Adding tests now